File: client/scalaz/src/main/scala/org/openapitools/client/api/IntervalList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import IntervalList._

case class IntervalList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Interval]])

object IntervalList {
  import DateTimeCodecs._

  implicit val IntervalListCodecJson: CodecJson[IntervalList] = CodecJson.derive[IntervalList]
  implicit val IntervalListDecoder: EntityDecoder[IntervalList] = jsonOf[IntervalList]
  implicit val IntervalListEncoder: EntityEncoder[IntervalList] = jsonEncoderOf[IntervalList]
}

File: API/Client/IntervalList.js
goog.provide('API.Client.IntervalList');

/**
 * @record
 */
API.Client.IntervalList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.IntervalList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.IntervalList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Interval>}
 * @export
 */
API.Client.IntervalList.prototype.data;

